lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -123,5 +156,19 @@ def find_latest_rs(deployment_data)
       rs.sync(latest_rs_data)
       rs
     end
+
+    def min_available_replicas
+      max_unavailable = @definition.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
```

Comment:
This one _could_ very well be server-side defaulted. IIRC you originally cached the entire resource blob from the `sync` to get this, and I asked you to save the specific value instead. Why are we not using it at all now? Am I forgetting something from earlier review passes?